### PR TITLE
Add person, student and teacher classes

### PR DESCRIPTION
**Pull request by elasfarc**
In this PR, I did the following:

Create a Person parent class
Create a Student class Inherits from Person
Create a Teacher class Inherits from Person

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  def can_use_services?
+    is_of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    age >= 18
+  end
```
Comment (ElijahTobs) [inline person.rb]: - [ ] You named the method on line 18 `of_age` but called `is_of_age` on line 13. Kindly make the necessary changes.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class Person
+  def initialize(age:, name: 'Unknown', parent_permission: true)
+    @id = nil
```
Comment (ElijahTobs) [inline person.rb]: - [ ] The `id` cannot be nil. You can generate a random number for this.

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required ⚠️ 🚦
## Hi @elasfarc 👋

You've done really well so far! 👍 

## Major Highlights 💯 

Your README is professional ✔️ 
Good commit messages ✔️ 
No linter errors ✔️ 
Pull Request has proper title and description ✔️ 

_However, there are some changes that you are required to make before this project can be approved._

## Required Changes ♻️

_See the inline comments below_

_If anything doesn't seem clear to you, feel free to leave your question or comment in the PR thread below_

#### "Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise."

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
## Happy Coding :computer: 

_Reviewed by_  **Elijah O. Ayandokun**

**Comment by Davidosky007** [comment]
## Hi @elasfarc 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 
good-job-gif



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**